pyteal/ast/subroutine.py

```diff
@@ -177,36 +184,50 @@ def _arg_types_and_by_refs(
             expected_arg_type = SubroutineDefinition._validate_parameter_type(
                 annotations, name
             )
-            expected_arg_types.append(expected_arg_type)
+
+            if param.kind is Parameter.KEYWORD_ONLY:
+                if not isinstance(expected_arg_type, abi.TypeSpec):
+                    raise TealInputError(
+                        f"Function keyword parameter {name} has type {expected_arg_type}"
+                    )
+                abi_output_kwarg[name] = expected_arg_type
+                continue
```

Comment:
@ahangsu As a I started reviewing, I made a few local changes.  I bundled them in a child PR for your consideration:  https://github.com/algorand/pyteal/pull/308.

Treat as optional.  Feel welcomed to tell me you'd prefer as is.

Reply:
@ahangsu Thanks for bringing in #308 - Resolving.